### PR TITLE
feat(settings): add JavaScript Source Maps support `.js.map` setup check

### LINKED ISSUE #20549: Add setup check for map files serving
> we should have a config check
> for serving map files
> because I'm sure tons of admins did not updated the apache/nginx configs

cc @rullzer

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/20549

### CODE REVIEW

File: apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		protected LoggerInterface $logger,
+	) {
+	}
+
+	public function getCategory(): string {
+		return 'network';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('JavaScript Source Maps support');
```
Comment (susnux) [inline apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php:56]:
```suggestion
		return $this->l10n->t('JavaScript source map support');
```

from other wording?

File: apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	public function run(): SetupResult {
+		$testFile = $this->urlGenerator->linkTo('settings', 'js/map-test.js.map');
+
+		foreach ($this->runHEAD($testFile) as $response) {
+			return SetupResult::success();
```
Comment (susnux) [inline apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php:63]: Do we need some specific content type?

File: apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		}
+
+		return SetupResult::error($this->l10n->t('Your webserver is not set up to serve `.js.map` files. Without these files, JavaScript Source Maps won\'t function properly, making it more challenging to troubleshoot and debug any issues that may arise.'));
```
Comment (come-nc) [inline apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php:66]: Maybe warning if it’s only for debugging?

File: apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php (hunk 2 above)
Comment (skjnldsv) [reply apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php:63]: As long as it's serving it, I think it's fine

File: apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php (hunk 3 above)
Comment (skjnldsv) [inline apps/settings/lib/SetupChecks/JavaScriptSourceMaps.php:66]:
```suggestion
		return SetupResult::warn($this->l10n->t('Your webserver is not set up to serve `.js.map` files. Without these files, JavaScript Source Maps won\'t function properly, making it more challenging to troubleshoot and debug any issues that may arise.'));
```

**Review by come-nc — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**